src/Type/ObjectType.php

```diff
@@ -66,8 +65,12 @@ public function hasProperty(string $propertyName): TrinaryLogic
 
 	public function getProperty(string $propertyName, ClassMemberAccessAnswerer $scope): PropertyReflection
 	{
-		$broker = Broker::getInstance();
-		return $broker->getClass($this->className)->getProperty($propertyName, $scope);
+		$classReflection = $this->getClassReflection();
+		if ($classReflection === null) {
+			throw new \PHPStan\ShouldNotHappenException();
```

Comment:
I think that original code throws `PHPStan\Broker\ClassNotFoundException` which is handled in a nicer way. Check out `catch (\PHPStan\AnalysedCodeException $e)` instances in Analyser.php.